server/save.js

```diff
@@ -17,7 +20,8 @@ exports.handleSave = function(req, res, app) {
   try {
     var user = res.locals.owner;
     var filename = utils.param(req, "file", utils.filenameFromUri(req));
-    var origfilename = filename;
+    var thumbname = path.join(path.dirname(filename), THUMB_DIR,
```

Comment:
It might be nice if computing the thumbnail file from a code file was a utility function. That could at least handle some of the usages here in save.js, but I'm not sure if there's a good way to handle the usage in load.js (since that one uses the absolute path). What do you think?